sphinx/domains/std.py

```diff
@@ -943,10 +943,17 @@ def _resolve_option_xref(self, env: "BuildEnvironment", fromdocname: str,
         progname = node.get('std:program')
         target = target.strip()
         docname, labelid = self.progoptions.get((progname, target), ('', ''))
-        # for :option:`-foo=bar` search for -foo option directive
-        if not docname and '=' in target:
-            target2 = target[:target.find('=')]
-            docname, labelid = self.progoptions.get((progname, target2), ('', ''))
+        if not docname:
+            # Support also reference that contain an option value:
+            # * :option:`-foo=bar`
+            # * :option:`-foo[=bar]`
+            # * :option:`-foo bar`
+            for needle in ('=', '[=', ' '):
+                idx = target.find(needle)
+                if idx != -1:
+                    docname, labelid = self.progoptions.get((progname, target[:idx]), ('', ''))
```

Comment:
More readable / 'pythonic', I think:
```suggestion
                if needle in target:
                    stem, _, _ = target.partition(needle)
                    docname, labelid = self.progoptions.get((progname, stem), ('', ''))
```